CONTRIBUTING.md

```diff
@@ -1,26 +1,101 @@
-# Dynatrace Operator
+- [Contributing](#contributing)
+  - [How to run unit tests](#how-to-run-unit-tests)
+  - [How to run E2E tests](#how-to-run-e2e-tests)
+  - [Coding style guide](#coding-style-guide)
+    - [General](#general)
+    - [Cuddling of statements](#cuddling-of-statements)
+    - [Reconciler vs Controller](#reconciler-vs-controller)
+      - [A **Controller** is a struct that **DIRECTLY** handles the reconcile Requests](#a-controller-is-a-struct-that-directly-handles-the-reconcile-requests)
+      - [A **Reconciler** is a struct that **INDIRECTLY** handles the reconcile Requests](#a-reconciler-is-a-struct-that-indirectly-handles-the-reconcile-requests)
+    - [Errors](#errors)
+      - [Do's](#dos)
+      - [Don'ts](#donts)
+    - [Logging](#logging)
+      - [Do's](#dos-1)
+      - [Don'ts](#donts-1)
+    - [Testing](#testing)
+      - [Do's](#dos-2)
+      - [Don'ts](#donts-2)
+  - [E2E testing guide](#e2e-testing-guide)
+  - [Useful commands](#useful-commands)
+    - [Remove all Dynatrace pods in force mode (useful debugging E2E tests)](#remove-all-dynatrace-pods-in-force-mode-useful-debugging-e2e-tests)
+    - [Copy CSI driver database to localhost for introspection via sqlite command](#copy-csi-driver-database-to-localhost-for-introspection-via-sqlite-command)
+    - [Add debug suffix on E2E tests to avoid removing pods](#add-debug-suffix-on-e2e-tests-to-avoid-removing-pods)
+    - [Debug cluster nodes by opening a shell prompt (details here)](#debug-cluster-nodes-by-opening-a-shell-prompt-details-here)
+
+# Contributing
+
+1. Fork the dynatrace-operator repository and get the source code:
+
+```sh
+git clone https://github.com/<your_username>/dynatrace-operator
+cd dynatrace-operator
+```
+
+2. Install development prerequisites:
+
+```sh
+make prerequisites
+```
+
+3. Create a new branch to work on:
+
+```sh
+git checkout -b feature/your-branch
+```
+
+4. Once the changes are finished, make sure there are no warnings on the code and [run the unit tests](#how-to-run-unit-tests) and [end-to-end tests](#how-to-run-e2e-tests) if necessary.
+
+```sh
+make go/test
+make test/e2e/<your_changes_scope>
+```
```

Comment:
It is worth to mention, that we have a pre-commit hook, so it's recommended to install it and it will run all unit tests and code checks:
```
make prerequisites/setup-pre-commit
```

Reply:
Done :+1:

Reply:
never mind I see it:
>`make prerequisites`